build.gradle

```diff
@@ -10,6 +11,7 @@ buildscript {
     dependencies {
         classpath 'com.diffplug.spotless:spotless-plugin-gradle:6.20.0'
         classpath 'gradle.plugin.com.hierynomus.gradle.plugins:license-gradle-plugin:0.16.1'
+        classpath 'com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.13.0'
```

Comment:
What does this do?

Reply:
this and above is a remnant of me trying to parse the metadata.yaml to only apply the extra checks to certified connectors. Instead, I'll be using the CDK version, which is simpler but means that stuff will start breaking whenever we try to upgrade the CDK version to 0.3.x

Reply:
This CDK version approach sounds rather brittle, in the meantime why not hardcode a whitelist of certified connectors? Of course it will grow stale, but that's not so bad, because we never downgrade connectors from certified status to non-certified. Until we certify another connector we should be able to come up with a decent way to parse these files. Feel free to create an issue to that effect and assign it to me.

Reply:
I just realized there's a couple more reasons why I went that road. I initially did parse the metadata.xml, and things were working but:
1) we would need to split this change into several: 1 to push the api exports into the new CDK version, then one to update all certified connectors to the latest CDK version, and finally one to enable the compiler checks. This would be necessary because the current CDK is missing package exports, which means all connectors have warnings.
2) I was also thinking that it'd be nice to be more stringent with new connectors, certified or not. The CDK-version handling gives us that. If you still prefer using the certified flag, I'll create the 3  required PRs

Reply:
Alternatively, I can just disable the -Werror for all the non-certified connectors, this way new connectors get the more stringent checks, and everything is simpler. That requires me to edit all the build.gradle, but I don't see that as a big deal

Reply:
Oh, I like that idea!